ministryofjustice/correspondence_tool_staff
# CT-2994 Fixed the search feature for kilo users

## Description

This PR is to fix the bug related the scope of the users including the kilo users can see FOI cases, the detail for explaining the permission/expectation for those can be checked from Jira ticket. 
Basically 
- FOI and Internal Review cases can be discovered by any user on the tool but subject to the scope which case type the user is allowed to see --> this is the FOI scope policy,
- For responder: "Open cases tab" and “Closed cases” only show the cases being assigned to them ---> this is functional feature, not related to scope policy at all
- Other case types are restricted to the role's permission per case type

The reason about why this issue happens is that I changed the scope of FOI policy by moving the restriction which was under Open/closed cases tab to policy level by misunderstanding/not knowing the rules and workflow. The solution I did is to move the responder restriction for specific pages back to the level where it should be.  I was thinking to expand the policy level by taking extra factor of which page the user is on, but I found this method will make the policy more vague (as fundamentally the FOI is public available) and also will add the extra parameter for all other policies.

## Self-review checklist

* [] (1) Quick stakeholder demo done OR

### Screenshots

None
### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-2994
### Deployment

No extra steps
### Manual testing instructions

1 - Login as a non-admin user from one of london teams
2 - create a FOI case without being assigned to any team, write down the case number
3 - Login as another non-admin user from one of london teams
4 - search the case being created at step2,  
5 - the case at step2 should be shown, you can click it to view the detail but no other operations are allowed
